From daf36df32d31cd298c1cb44419f67f8676a4f008 Mon Sep 17 00:00:00 2001
From: Darek Stojaczyk <dariusz.stojaczyk@intel.com>
Date: Tue, 30 Jun 2020 11:24:24 +0200
Subject: [PATCH 2/3] build/meson: disable qat_asym driver

QAT docs in DPDK say:
[...] for crypto, where each QAT VF may expose two crypto devices,
sym and asym, it may happen that the number of devices will be bigger
than MAX_DEVS and the process will show an error during PMD
initialisation. To avoid this problem CONFIG_RTE_CRYPTO_MAX_DEVS may
be increased or -w, pci-whitelist domain:bus:devid:func option may be
used.

We run into this limitation in our CI. We didn't compile the asym driver
with Makefiles, so don't compile it with meson either. Increasing
MAX_DEVS unnecessarily increases the total memory footprint, so we
don't go that way.

Change-Id: Idf27335ef0c8890727909985bd3e909edb1282e6
Signed-off-by: Darek Stojaczyk <dariusz.stojaczyk@intel.com>
Signed-off-by: Tomasz Zawadzki <tomasz.zawadzki@intel.com>
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/3517 (spdk-20.05)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/3770 (spdk-20.08)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/5338 (spdk-20.11)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/6428
Reviewed-by: Jim Harris <james.r.harris@intel.com>
Reviewed-by: Ben Walker <benjamin.walker@intel.com>
---
 drivers/common/qat/meson.build | 4 ++--
 drivers/crypto/qat/meson.build | 5 +----
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/drivers/common/qat/meson.build b/drivers/common/qat/meson.build
index 67f7aca974..6d10fcc36a 100644
--- a/drivers/common/qat/meson.build
+++ b/drivers/common/qat/meson.build
@@ -54,10 +54,10 @@ endif
 
 if qat_crypto
 	foreach f: ['qat_sym_pmd.c', 'qat_sym.c', 'qat_sym_session.c',
-			'qat_sym_hw_dp.c', 'qat_asym_pmd.c', 'qat_asym.c']
+			'qat_sym_hw_dp.c']
 		sources += files(join_paths(qat_crypto_relpath, f))
 	endforeach
 	deps += ['security']
 	ext_deps += libcrypto
-	cflags += ['-DBUILD_QAT_SYM', '-DBUILD_QAT_ASYM']
+	cflags += ['-DBUILD_QAT_SYM']
 endif
diff --git a/drivers/crypto/qat/meson.build b/drivers/crypto/qat/meson.build
index 92e0ed6565..bad65dc3a3 100644
--- a/drivers/crypto/qat/meson.build
+++ b/drivers/crypto/qat/meson.build
@@ -15,10 +15,7 @@ if dep.found()
 	qat_sources += files('qat_sym_pmd.c',
 			     'qat_sym.c',
 			     'qat_sym_session.c',
-			     'qat_sym_hw_dp.c',
-			     'qat_asym_pmd.c',
-			     'qat_asym.c')
+			     'qat_sym_hw_dp.c')
 	qat_ext_deps += dep
 	qat_cflags += '-DBUILD_QAT_SYM'
-	qat_cflags += '-DBUILD_QAT_ASYM'
 endif
-- 
2.26.2

